File: funcj/src/de/fuh/seminar1908/funcjp/Library.java
package de.fuh.seminar1908.funcjp;

import java.time.Year;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

public class Library {
    public static List<Book> getBooks() {
        List<Book> books = new ArrayList<>();

        // Book functThinking = new Book();
        // functThinking.setTitle("Functional Thinking");
        // functThinking.setAuthors(Arrays.asList("Neal Ford"));
        // functThinking.setYear(Year.of(2014));
        // books.add(functThinking);

        Book j8InAction = new Book();
        j8InAction.setTitle("Java 8 in Action");
        j8InAction.setAuthors(Arrays.asList("Raoul-Gabriel Urma", "Mario Fusco",
                "Alan Mycroft"));
        j8InAction.setYear(Year.of(2014));
        books.add(j8InAction);

        Book j8Lambdas = new Book();
        j8Lambdas.setTitle("Java 8 Lambdas");
        j8Lambdas.setAuthors(Arrays.asList("Richard Warburton"));
        j8Lambdas.setYear(Year.of(2014));
        books.add(j8Lambdas);

        Book fluentPython = new Book();
        fluentPython.setTitle("Fluent Python");
        fluentPython.setAuthors(Arrays.asList("Luciano Ramalho"));
        fluentPython.setYear(Year.of(2016));
        books.add(fluentPython);

        return books;
    }
}

File: funcp/src/de/fuh/seminar1908/funcjp/library.py
class Book:
    title = None
    authors = []
    year = None
    
    def __repr__(self):
        return f"Book({self.title})"


j8_in_action = Book()
j8_in_action.title = "Java 8 in Action"
j8_in_action.authors = ["Raoul-Gabriel Urma", "Mario Fusco",
                        "Alan Mycroft"]
j8_in_action.year = 2014

j8_lambdas = Book()
j8_lambdas.title = "Java 8 Lambdas"
j8_lambdas.authors = ["Richard Warburton"]
j8_lambdas.year = 2014

fluent_python = Book()
fluent_python.title = "Fluent Python"
fluent_python.authors = ["Luciano Ramalho"]
fluent_python.year = 2016

books = [j8_in_action, j8_lambdas, fluent_python]